dandi/cli/cmd_validate.py

```diff
@@ -111,50 +117,44 @@ def validate(
         devel_debug=devel_debug,
         allow_any_path=allow_any_path,
     )
+    _process_issues(validator_result, grouping, ignore)
 
-    _process_issues(validator_result, grouping)
-
-
-def _process_issues(validator_result, grouping):
 
-    issues = [i for i in validator_result if i.severity]
-
-    purviews = [
-        list(filter(bool, [i.path, i.path_regex, i.dataset_path]))[0] for i in issues
-    ]
+def _process_issues(
+    validator_result: Iterable[ValidationResult], grouping: str, ignore: Optional[str]
+) -> None:
+    issues = [i for i in validator_result if i.severity is not None]
+    if ignore is not None:
+        issues = [i for i in issues if not re.search(ignore, i.id)]
+    purviews = [i.purview for i in issues]
     if grouping == "none":
         display_errors(
             purviews,
             [i.id for i in issues],
-            [i.severity for i in issues],
+            cast(List[Severity], [i.severity for i in issues]),
             [i.message for i in issues],
         )
     elif grouping == "path":
         for purview in purviews:
-            applies_to = [
-                i for i in issues if purview in [i.path, i.path_regex, i.dataset_path]
-            ]
+            applies_to = [i for i in issues if purview == i.purview]
             display_errors(
                 [purview],
                 [i.id for i in applies_to],
-                [i.severity for i in applies_to],
+                cast(List[Severity], [i.severity for i in applies_to]),
```

Comment:
me is wondering why this casting is needed since it seems it is explicit that it would be a list of Severities.

Reply:
`severity` can be `None`, so mypy sees `[i.severity for i in applies_to]` as a `list[Optional[Severity]]`, yet we've filtered out the `None` severities above.